lib/spack/spack/build_systems/intel.py

```diff
@@ -103,6 +135,711 @@ def _filtered_components(self):
+
+        if component == 'mpi' or 'intel-mpi' in self.name:
+            d = d.intel64
+        d = d.include
+
+        if relative:
+            d = self.relative_path(d)
+
+        debug_print(d)
+        return d
+
+    @property
+    def omp_libs(self):
+        # Supply LibraryList for linking OpenMP
+
+        # FIXME  if sys.platform == 'darwin' ....
+
+        # TODO?  Across Spack packages for "client" applications, are variants
+        #   named consistently enough to determine if OpenMP is even needed?
+
+        if '%intel' in self.spec:
+            omp_libnames = ['libiomp5']
+
+            # Note about search root: For MKL, the directory
+            # "$MKLROOT/../compiler" will be present even for an MKL-only
+            # product installation (as opposed to one being ghosted via
+            # packages.yaml), specificially to provide the 'iomp5' libs.
+
+            omp_libs = find_libraries(
+                omp_libnames,
+                root=self.component_lib_dir(component='compiler'),
+                shared=self._want_shared)
+
+        elif '%gcc' in self.spec:
+            gcc = Executable(self.compiler.cc)
+            omp_libnames = gcc('--print-file-name',
+                               'libgomp.{0}'.format(dso_suffix),
+                               output=str)
+            omp_libs = LibraryList(omp_libnames)
+
+        if len(omp_libs) < 1:
+            raise_lib_error('Cannot locate OpenMP libraries:', omp_libnames)
+
+        debug_print(omp_libs)
+        return omp_libs
+
+    @property
+    def blas_libs(self):
+        # Main magic here.
+        # For reference, see The Intel Math Kernel Library Link Line Advisor:
+        # https://software.intel.com/en-us/articles/intel-mkl-link-line-advisor/
+
+        mkl_libnames = ['libmkl_intel_' + self.intel64_int_suffix]
+        if not self.spec.satisfies('threads=openmp'):
+            mkl_libnames.append('libmkl_sequential')
+        elif '%intel' in self.spec:
+            mkl_libnames.append('libmkl_intel_thread')
+        elif '%gcc' in self.spec:
+            mkl_libnames.append('libmkl_gnu_thread')
+        else:
+            raise_lib_error('Cannot determine MKL threading libraries.')
+        mkl_libnames.append('libmkl_core')
+
+        mkl_libs = find_libraries(
+            mkl_libnames,
+            root=self.component_lib_dir(component='mkl'),
+            shared=self._want_shared)
+        debug_print(mkl_libs)
+
+        if len(mkl_libs) < 3:
+            raise_lib_error('Cannot locate core MKL libraries:', mkl_libnames)
+
+        # TODO? TBB threading: ['libmkl_tbb_thread', 'libtbb', 'libstdc++']
+        #
+        # NB: TBB is C++, and I don't think here is the right place to try
+        # messing with compiler specifics to spit out the right libtbb and
+        # libstdc++. Can't a user now use the Spack 'tbb' provider concept?
+
+        if self.spec.satisfies('threads=openmp'):
+            omp_libs = self.omp_libs()
```

Comment:
this is used in one place only. I think you make `blas_libs` much harder to read with this logic. Would you mind simply using the original code here with a few extra bits and pieces related to parent directories?

Also that `if self.spec.satisfies('threads=openmp'):` should actually be joined with those ifs in the above.

Reply:
> How's that?

LGTM